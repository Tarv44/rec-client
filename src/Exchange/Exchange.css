header {
    display: flex;
    flex-direction: column;
}

#exchange-header {
    border-bottom: solid black 3px;
    color: #454545
}

#exchange-header *{
    border-bottom: #0CF0AB solid 3px;
}

#exchange-title {
    font-size: 30px;
}

#ex-add-song-btn {
    padding: 2em 0;
    letter-spacing: .3em;
    font-weight: bold;
    color: black;
}

#ex-add-song {
    padding: 3%;
    display: flex;
    flex-direction: column;
}

#ex-submit-song-btn {
    align-self: center;
    margin-top: .5em;
    padding: .5em 1.5em;
    font-size: 16px;
}