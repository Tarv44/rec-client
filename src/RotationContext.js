import React from 'react';

const RotationContext = React.createContext({
    updateUser: () => {}
})

export default RotationContext;