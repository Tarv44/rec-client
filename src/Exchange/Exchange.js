import { Component } from 'react';
import Song from '../Song/Song';
import RotationContext from '../RotationContext';
import './Exchange.css';

export default class Exchange extends Component {
    static contextType = RotationContext

    constructor(props) {
        super(props)
        this.state = {
            id: Number(this.props.match.params.exId),
            title: '',
            date_created: '',
            created_by: '',
            songs: []
        }
    }

    componentDidMount() {
        const store = this.context
        const exchange = store.exchanges.find(ex => ex.id === this.state.id)
        const created_by = store.users.find(user => user.id === exchange.created_by).username
        const songs = store.songs.filter(song => song.exchange_id === this.state.id)
            .map(song => {
                const songComments = store.comments.filter(comment => comment.song_id === song.id)
                    .map(comm => {
                        return {
                            ...comm,
                            created_by: store.users.find(user => user.id === comm.created_by).username
                        }
                    })
                song.comments = songComments
                return song
            })
        this.setState({
            title: exchange.title,
            date_created: exchange.date_created.toLocaleDateString(),
            created_by,
            songs
        })
    }

    render() {
        const songs = this.state.songs.map((song, i) => {
            return <Song song={song} key={i}/>
        })
        return (
            <main>
                <header>
                    <h2>{this.state.title}</h2>
                    <h5>Created by {this.state.created_by} on {this.state.date_created}</h5>
                </header>
                {songs}
            </main>
        )
    }
}