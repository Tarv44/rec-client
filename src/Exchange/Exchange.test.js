import React from 'react'
import ReactDom from 'react-dom'
import renderer from 'react-test-renderer'
import Exchange from './Exchange'
import { BrowserRouter } from 'react-router-dom'
import RotationContext from '../RotationContext'


describe('----- Exchange -----', () => {
    const match = {
        params: {
            exId: 1
        }
    }

    const context = {
        current_user: {
            id: 1,
            username: 'user'
        }
    }
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDom.render((
            <BrowserRouter>
                <RotationContext.Provider value={context}>
                    <Exchange match={match}/>
                </RotationContext.Provider>
            </BrowserRouter> 
        ), div)
        ReactDom.unmountComponentAtNode(div);
    })
    it('renders the UI as expected', () => {
        const comp = renderer
            .create(
                <BrowserRouter>
                    <RotationContext.Provider value={context}>
                        <Exchange match={match}/>
                    </RotationContext.Provider>
                </BrowserRouter> 
            )
            .toJSON()
        expect(comp).toMatchSnapshot()
    })
})