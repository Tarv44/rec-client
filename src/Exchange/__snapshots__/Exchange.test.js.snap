// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`----- Exchange ----- renders the UI as expected 1`] = `
<main
  id="loading-screen"
>
  <img
    alt="Rotation logo"
    id="loading-logo"
    src="logo.png"
  />
  <h3
    id="loading-message"
  >
    Queing the tracks
    ...
  </h3>
</main>
`;
